### PR TITLE
ci: add lab branch to publish workflow

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/docker/docker.github.io/pull/15151

We are working with @VictorBersy to add a new environment to test advanced features on docs repo without impacting staging (https://docs-stage.docker.com/) and production (https://docs.docker.com/) environment.

This environment will be published to `https://docs-lab.docker.com/`.

I have created a lab branch where we are going to make some tests with https://github.com/docker/docker.github.io/pull/15151.

This PR adds the necessary configuration to our publish workflow to allow manual invocations on lab branch (see last commit).

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

**Comment by thaJeztah** [comment]
qq; do we still need "force pushes" on the `lab` branch protection? Or was that only for initial testing?

**Comment by crazy-max** [comment]
> qq; do we still need "force pushes" on the `lab` branch protection? Or was that only for initial testing?

was temporary yes, I just turned off the force pushes.

File: _releaser/_website-config-docs-lab.json [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+  "IndexDocument": {
+    "Suffix": "index.html"
+  },
+  "RedirectAllRequestsTo": null,
+  "RoutingRules": [
+    {
+      "Condition": {
+        "HttpErrorCodeReturnedEquals": null,
+        "KeyPrefixEquals": "v1.4/"
+      },
```
Comment (thaJeztah) [inline _releaser/_website-config-docs-lab.json:14]: Do you think we should consider to "skip" these redirects for the labs domain? (I'm fine either way; mostly thinking; we're probably won't be using these on that domain - or are we testing against them?)

Comment (crazy-max) [reply _releaser/_website-config-docs-lab.json:14]: Yes in a follow-up but for now we keep them as @VictorBersy will first create the lab environment so we can test everything is ok with the current workflow and after that we take into account https://github.com/docker/docker.github.io/pull/15151

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -15,7 +17,11 @@ jobs:
         run: |
           JEKYLL_ENV=development
           DOCS_AWS_REGION=us-east-1
-          if [ "${{ github.ref }}" = "refs/heads/master" ]; then
+          if [ "${{ github.ref }}" = "refs/heads/lab" ]; then
+            DOCS_S3_BUCKET=""
+            DOCS_LAMBDA_FUNCTION_CACHE=""
+            DOCS_S3_CONFIG="_website-config-docs-lab.json"
+          elif [ "${{ github.ref }}" = "refs/heads/master" ]; then
```
[the remark this answers is not in the document]
Comment (crazy-max) [reply .github/workflows/publish.yml]: Done

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Pull request merged**